### PR TITLE
rfb: add RFB frames

### PR DESCRIPTION
**Pull request by hsadia538**
- [X] I have read the contributing guide lines at https://suricata.readthedocs.io/en/latest/devguide/codebase/contributing/contribution-process.html
- [X] I have signed the Open Information Security Foundation contribution agreement at https://suricata.io/about/contribution-agreement/
- [ ] I have updated the user guide (in doc/userguide/) to reflect the changes made (if applicable)

Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket: https://redmine.openinfosecfoundation.org/issues/5717
 
Previous PR: #8339 

Describe changes:
- Fix format issues
- Fix commit issues

suricata-verify-pr: 1050

### CODE REVIEW

File: rust/src/rfb/rfb.rs [hunk 1]
```diff
@@ -226,7 +243,11 @@ impl RFBState {
                             consumed += current.len() - rem.len();
                             current = rem;
 
-                            self.state = parser::RFBGlobalState::TCSecurityResult;
+                            if let Some(frame) = &pdu {
+                                frame.set_len(flow, consumed as i64);
+                            }
+
+                        self.state = parser::RFBGlobalState::TCSecurityResult;
```
Comment (inashivb) [inline rust/src/rfb/rfb.rs:255]: nit: indent

File: rust/src/rfb/rfb.rs [hunk 2]
```diff
@@ -248,7 +269,11 @@ impl RFBState {
                             consumed += current.len() - rem.len();
                             current = rem;
 
-                            self.state = parser::RFBGlobalState::TCServerInit;
+                            if let Some(frame) = &pdu {
+                                frame.set_len(flow, consumed as i64);
+                            }
+
+                        self.state = parser::RFBGlobalState::TCServerInit;
```
Comment (inashivb) [inline rust/src/rfb/rfb.rs:282]: nit: indent

File: rust/src/rfb/rfb.rs [hunk 3]
```diff
@@ -388,7 +428,11 @@ impl RFBState {
                             consumed += current.len() - rem.len();
                             current = rem;
 
-                            self.state = parser::RFBGlobalState::TSVncResponse;
+                            if let Some(frame) = &pdu {
+                                frame.set_len(flow, consumed as i64);
+                            }
+
+                        self.state = parser::RFBGlobalState::TSVncResponse;
```
Comment (inashivb) [inline rust/src/rfb/rfb.rs:462]: nit: indent

**Review by inashivb — CHANGES_REQUESTED** [review]
Looking ok to me in the preliminary analysis. But, given my limited knowledge on the topic, I'll pass on to @jufajardini to review it further.

File: rust/src/rfb/rfb.rs (hunk 1 above)
Comment (hsadia538) [reply rust/src/rfb/rfb.rs:255]: Same issue as I mentioned yesterday. The commit appears differently in my editor where this line has a different indent level.

**Review by jufajardini — COMMENTED** [review]
This is looking good, nice work! :)
I have left an inline comment about the frame types.
And we must figure out what's going on with the indentation issues...

File: rust/src/rfb/rfb.rs [hunk 4]
```diff
@@ -22,11 +22,17 @@ use std::ffi::CString;
...
 use nom7::Err;
 use super::parser;
 
 static mut ALPROTO_RFB: AppProto = ALPROTO_UNKNOWN;
 
+#[derive(AppLayerFrameType)]
+pub enum RFBFrameType {
+    Pdu,
+}
+
```
[the remark this answers is not in the document]
Comment (hsadia538) [reply rust/src/rfb/rfb.rs:35]: do you mean e.g client vs server frames maybe?

File: rust/src/rfb/rfb.rs (hunk 1 above)
Comment (hsadia538) [reply rust/src/rfb/rfb.rs:255]: I think this was happening due to formatting in the wrong commit. I will fix it in the next PR

File: rust/src/rfb/rfb.rs (hunk 4 above)
[the remark this answers is not in the document]
Comment (jufajardini) [reply rust/src/rfb/rfb.rs:35]: More like VNC Challenge, or Security Result, for instance. But this is a good example of an aspect that needs input from more folks.

**Comment by hsadia538** [comment]
Followed by #8358

**Pull request closed without merging**